### PR TITLE
metadata for release `0.16.0` (`24Q4_B15`)

### PR DESCRIPTION
**Pull request by zacharyburnett**
- [x] freeze entry in `CHANGES.rst`
- [x] add entry to build table in `README.md`
- [x] update dependency pins in `pyproject.toml` and remove all `git+` dependency specifiers

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -25,13 +25,13 @@ dependencies = [
...
+    "roman_datamodels>=0.21.0",
+    # "roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git",
     "scipy >=1.11",
-    #"stcal>=1.7.3,<1.8.0",
-    "stcal @ git+https://github.com/spacetelescope/stcal.git@main",
-    #"stpipe >=0.5.0",
-    "stpipe @ git+https://github.com/spacetelescope/stpipe.git@main",
+    "stcal>=1.7.4,<1.8.0",
+    # "stcal @ git+https://github.com/spacetelescope/stcal.git@main",
+    "stpipe >=0.7.0",
```
Comment (braingram) [inline pyproject.toml:33]:
```suggestion
    "stpipe >=0.7.0,<0.8.0",
```

File: pyproject.toml [hunk 2]
```diff
@@ -25,13 +25,13 @@ dependencies = [
     "requests >=2.26",
     "rad>=0.21.0",
     # "rad @ git+https://github.com/spacetelescope/rad.git",
-    # "roman_datamodels>=0.21.0",
-    "roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git",
+    "roman_datamodels>=0.21.0",
```
Comment (braingram) [inline pyproject.toml:28]:
```suggestion
    "roman_datamodels>=0.21.0,<0.22.0",
```

File: pyproject.toml [hunk 3]
```diff
@@ -25,13 +25,13 @@ dependencies = [
     "requests >=2.26",
     "rad>=0.21.0",
```
Comment (braingram) [inline pyproject.toml:26]: upper pin here too?

**Review by braingram — COMMENTED** [review]
Maybe I took the pin suggestions a bit far but I'm thinking upper pins for:
- rad
- roman_datamodels
- stpipe (to match jwst)

I expect these will all be relaxed back to pointing to main for rad and roman_datamodels as development continues. I think these upper pins should allow the dev versions to work (until the next release). For example rad (when a new commit is added) will be at `'0.21.1.dev` so the dev tests will still satisfy these pins.

Let me know if this makes sense.

File: pyproject.toml (hunk 3 above)
Comment (zacharyburnett) [reply pyproject.toml:26]:
```suggestion
    "rad>=0.21.0,<0.22.0",
```

**Comment by zacharyburnett** [comment]
that sounds good to me

File: README.md [hunk 4]
```diff
@@ -167,6 +167,7 @@ an [issue](https://github.com/spacetelescope/romancal/issues).
 | 0.14.0    | 24Q2_B13  | 056          | Feb 2024  | Release for Build 24Q2_B13 (Build 13) |
 | 0.15.0    | 24Q3_B14  | 058          | May 2024  | Release for Build 24Q3_B14 (Build 14) |
 | 0.15.1    | 24Q3_B14  | 058          | May 2024  | Release for Build 24Q3_B14 (Build 14) |
+| 0.16.0    | 24Q4_B15  | 061          | Sep 2024  | Release for Build 24Q3_B15 (Build 15) |
```
Comment (zacharyburnett) [inline README.md:170]: should the Date column be the release date (could be August?) or build date (September)?

Comment (ddavis-stsci) [reply README.md:170]: I think our release 0.16.0 should be when we put the release together, so Aug 2024
When our release is bundled with the SDF code they can use the Sept 2024 date. 
Also can you update the context from 061 --> 063

**Review by ddavis-stsci — APPROVED** [review]

**Pull request merged**